### ISSUE TITLE
Remove dependency on http://www.csie.ntu.edu.tw for LibSVMIter test

### ISSUE THREAD

**Issue by anirudh2290** [apache/incubator-mxnet#8542]
Note: Providing complete information in the most concise form is the best way to get help. This issue template serves as the checklist for essential information to most of the technical issues.

If the issue is non-technical, feel free to present the information in what you believe is the best form.

## Description
The dependency here on www.csie.ntu.edu.tw should be changed to be hosted on our own S3 bucket.  See https://github.com/apache/incubator-mxnet/blob/master/tests/python/unittest/test_io.py#L218

**Comment by eric-haibin-lin** [comment]
We need to review the license of that dataset before hosting it ourselves.

**Comment by KellenSunderland** [comment]
Just had a test failure due to this external dep: http://jenkins.mxnet-ci.amazon-ml.com/blue/organizations/jenkins/incubator-mxnet/detail/PR-9055-merge/11/pipeline/337

```
======================================================================

ERROR: test_io.test_LibSVMIter

----------------------------------------------------------------------

Traceback (most recent call last):

  File "C:\Anaconda3\envs\py3\lib\site-packages\nose\case.py", line 197, in runTest

    self.test(*self.arg)

  File "C:\jenkins_slave\workspace\ut-python-cpu@2\tests\python\unittest\test_io.py", line 254, in test_LibSVMIter

    check_libSVMIter_news_data()

  File "C:\jenkins_slave\workspace\ut-python-cpu@2\tests\python\unittest\test_io.py", line 238, in check_libSVMIter_news_data

    news_metadata['origin_name'])

  File "C:\jenkins_slave\workspace\ut-python-cpu@2\pkg_vc14_cpu\python\mxnet\test_utils.py", line 1498, in get_bz2_data

    download(url, fname=data_origin_name, dirname=data_dir, overwrite=False)

  File "C:\jenkins_slave\workspace\ut-python-cpu@2\pkg_vc14_cpu\python\mxnet\test_utils.py", line 1411, in download

    r = requests.get(url, stream=True)

  File "C:\Anaconda3\envs\py3\lib\site-packages\requests\api.py", line 70, in get

    return request('get', url, params=params, **kwargs)

  File "C:\Anaconda3\envs\py3\lib\site-packages\requests\api.py", line 56, in request

    return session.request(method=method, url=url, **kwargs)

  File "C:\Anaconda3\envs\py3\lib\site-packages\requests\sessions.py", line 488, in request

    resp = self.send(prep, **send_kwargs)

  File "C:\Anaconda3\envs\py3\lib\site-packages\requests\sessions.py", line 630, in send

    history = [resp for resp in gen] if allow_redirects else []

  File "C:\Anaconda3\envs\py3\lib\site-packages\requests\sessions.py", line 630, in <listcomp>

    history = [resp for resp in gen] if allow_redirects else []

  File "C:\Anaconda3\envs\py3\lib\site-packages\requests\sessions.py", line 190, in resolve_redirects

    **adapter_kwargs

  File "C:\Anaconda3\envs\py3\lib\site-packages\requests\sessions.py", line 609, in send

    r = adapter.send(request, **kwargs)

  File "C:\Anaconda3\envs\py3\lib\site-packages\requests\adapters.py", line 497, in send

    raise SSLError(e, request=request)

requests.exceptions.SSLError: EOF occurred in violation of protocol (_ssl.c:749)

-------------------- >> begin captured logging << --------------------

requests.packages.urllib3.connectionpool: DEBUG: Starting new HTTP connection (1): www.csie.ntu.edu.tw

requests.packages.urllib3.connectionpool: DEBUG: http://www.csie.ntu.edu.tw:80 "GET /~cjlin/libsvmtools/datasets/multiclass/news20.t.bz2 HTTP/1.1" 301 287

requests.packages.urllib3.connectionpool: DEBUG: Starting new HTTPS connection (1): www.csie.ntu.edu.tw

--------------------- >> end captured logging << ---------------------
```

**Comment by eric-haibin-lin** [comment]
Fixed by #9620
